server/lib/object-storage/shared/client.ts

```diff
@@ -1,8 +1,23 @@
+import { NodeHttpHandler } from "@aws-sdk/node-http-handler"
+import { HttpsProxyAgent } from 'https-proxy-agent'
+import { getProxy, isProxyEnabled } from '@server/helpers/proxy'
 import { S3Client } from '@aws-sdk/client-s3'
 import { logger } from '@server/helpers/logger'
 import { CONFIG } from '@server/initializers/config'
 import { lTags } from './logger'
 
+function getProxyRequestHandler () {
+
+  if (!isProxyEnabled()) { return new NodeHttpHandler() }
+
+  const proxy = getProxy()
+
+  return new NodeHttpHandler({
+    httpAgent: new HttpsProxyAgent(proxy),
```

Comment:
Could you use `hpagent` instead? https://github.com/Chocobozzz/PeerTube/blob/develop/server/helpers/requests.ts#L165